api/envoy/config/core/v3/resolver.proto

```diff
@@ -14,11 +14,28 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: Resolver]
 
-// DNS resolver configuration which includes the underlying dns resolver addresses and options.
-message DnsResolver {
-  // A list of dns resolver addresses
+// Configuration of DNS resolver option flags which control the behavior of the DNS resolver.
+message DnsResolverOptions {
+  // Use TCP for all DNS queries instead of the default protocol UDP.
+  // Setting this value causes failure if the
+  // ``envoy.restart_features.use_apple_api_for_dns_lookups`` runtime value is true during
+  // server startup. Apple's API only uses UDP for DNS resolution.
+  bool use_tcp_for_dns_lookups = 1;
+
+  // Do not use the default search domains; only query hostnames as-is or as aliases.
+  bool no_default_search_domain = 2;
```

Comment:
I think you should add a version history entry for this change.

Reply:
Done, thanks.